### ISSUE TITLE
Missing Node modules when restoring Yarn cache

### ISSUE THREAD

**Issue by hdodov** [actions/cache#248]
I have a setup with Yarn where I have my main dependencies in the root `package.json`. I also have a submodule in my `src` folder and I have to install its dependencies as well before building my project.

When the dependencies are installed, my workflow works. However, when they are restored from cache, some of them are missing and the build fails. Specifically, [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin) is missing. Here are my workflow steps:

```yml
- name: Yarn - get cache directory
  id: yarn-cache-path
  run: echo "::set-output name=dir::$(yarn cache dir)"

- name: Yarn - process cache
  id: yarn-cache
  uses: actions/cache@v1
  with:
    path: ${{ steps.yarn-cache-path.outputs.dir }}
    key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
    restore-keys: ${{ runner.os }}-yarn-

- name: Yarn - install
  if: steps.yarn-cache.outputs.cache-hit != 'true'
  run: yarn install && cd ./src/js/oblik && yarn install --production=true

- name: Build
  run: yarn build
```

This is the command that installs the dependencies before they are cached:

```
yarn install && cd ./src/js/oblik && yarn install --production=true
```

Basically, the cache is correctly restored, but some modules seem to be missing. Perhaps that `--production` flag causes some issues? Here's the [workflow.log](https://github.com/actions/cache/files/4444163/workflow.log)

**Comment by zRelux** [comment]
--production will only install the regular deps and not the dev ones maybe that is the problem?

**Comment by hdodov** [comment]
@zRelux that's what I thought at first as well, but the missing dependency is from the first `yarn install` that doesn't have the `--production` flag. The second package doesn't have that dependency at all. I tried without `--production` and it still failed.

**Comment by dahu33** [comment]
You need to remove `if: steps.yarn-cache.outputs.cache-hit != 'true'`
